### PR TITLE
[feat]: implement pokemons-page/pagination

### PR DESCRIPTION
**Pull request by nbcjoke**

### CODE REVIEW

File: src/components/Header/index.js [hunk 1]
```diff
@@ -10,39 +10,37 @@ import { LoginContainer } from "../../pages/login/containers/LoginContainer";
...
+              <NavLink className="nav__wrapper-link" to="../products">
                 Products
-              </a>
+              </NavLink>
             </li>
             <li className="nav__wrapper-list">
-              <a href="#" className="nav__wrapper-link">
+              <NavLink className="nav__wrapper-link" to="../contact">
                 Contact us
-              </a>
```
Comment (adryianSiarheyeu) [inline src/components/Header/index.js:45]: use configs for static similar data

File: src/components/Header/index.js [hunk 2]
```diff
@@ -10,39 +10,37 @@ import { LoginContainer } from "../../pages/login/containers/LoginContainer";
 
 import { AccountCircle, ShoppingCart } from "@mui/icons-material";
 import { IconButton } from "@mui/material";
+
 import "./Header.css";
```
Comment (adryianSiarheyeu) [inline src/components/Header/index.js:14]: use css/scss modules

File: src/index.js [hunk 3]
```diff
@@ -9,19 +9,26 @@ import { Provider } from "react-redux";
 import Header from "./components/Header";
 import { Footer } from "./components/Footer";
 import "./reset.css";
+import { createTheme, ThemeProvider } from "@mui/material";
 
 const store = configureStore();
 
-window.React = React;
```
Comment (adryianSiarheyeu) [inline src/index.js:15]: wtf it was?

File: src/index.js [hunk 4]
```diff
@@ -9,19 +9,26 @@ import { Provider } from "react-redux";
 import Header from "./components/Header";
 import { Footer } from "./components/Footer";
 import "./reset.css";
```
Comment (adryianSiarheyeu) [inline src/index.js:11]: use modules. fix it in all places

File: src/index.js [hunk 5]
```diff
@@ -9,19 +9,26 @@ import { Provider } from "react-redux";
...
-        <Header />
-        <Router />
-        <Footer />
-      </HashRouter>
+      <ThemeProvider theme={theme}>
+        <HashRouter>
+          <Header />
+          <Router />
+          <Footer />
+        </HashRouter>
```
Comment (adryianSiarheyeu) [inline src/index.js:30]: use separated layout component

File: src/pages/PokemonDetails/containers/PokemonDetailsContainer.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+export const PokemonDetailsContainer = () => {
+  const dispatch = useDispatch();
+  const { pokemonName } = useParams();
+
+  const { info, isLoading } = useSelector(pokemonDetailsSelector);
+
+  useEffect(() => {
+    dispatch(GET_POKEMON_DETAILS_REQUEST(pokemonName));
+  }, []);
+  return <div>{isLoading ? "loading..." : <h1>{info.name}</h1>}</div>;
```
Comment (adryianSiarheyeu) [inline src/pages/PokemonDetails/containers/PokemonDetailsContainer.js]: create loading component and use it through the app

File: src/pages/Pokemons/components/PokemonsLayout/index.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import { Link } from "react-router-dom";
+
+import { ROUTE_NAMES } from "../../../../routes/routeNames";
+import { Pagination } from "../../../../components/Pagination";
```
Comment (adryianSiarheyeu) [inline src/pages/Pokemons/components/PokemonsLayout/index.js:4]: fix import order

File: src/pages/Pokemons/components/PokemonsLayout/index.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+                  </Link>
+                </Grid>
+                <Grid item>
+                  <Typography
+                    className="card__wrapper-price"
+                    gutterBottom
+                    variant="h6"
+                    component="div"
+                  >
+                    {price}$
```
Comment (adryianSiarheyeu) [inline src/pages/Pokemons/components/PokemonsLayout/index.js]: create util function which will format currency values

File: src/pages/Pokemons/components/PokemonsLayout/index.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+                  </Typography>
+                </Grid>
+              </Grid>
+              <Typography className="card__wrapper-info" variant="body2">
+                Pinstriped cornflower blue cotton blouse takes you on a walk to
+                the park or just down the hall.
+              </Typography>
```
Comment (adryianSiarheyeu) [inline src/pages/Pokemons/components/PokemonsLayout/index.js]: what the purpose of this dummy static text?

File: src/pages/Pokemons/components/PokemonsLayout/index.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+                <Grid item>
+                  <Typography
+                    className="card__wrapper-price"
+                    gutterBottom
+                    variant="h6"
+                    component="div"
+                  >
+                    {price}$
+                  </Typography>
+                </Grid>
```
Comment (adryianSiarheyeu) [inline src/pages/Pokemons/components/PokemonsLayout/index.js]: this must be separated component (PokemonCard), not bunch of useless markup

File: src/pages/Pokemons/components/PokemonsLayout/index.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+            </Grid>
+            <Box className="card__wrapper-buttons">
+              <Button color="primary" variant="outlined">
+                Buy
+              </Button>
+              <IconButton>
+                <ShoppingCart color="primary" />
+              </IconButton>
+            </Box>
+          </Box>
```
Comment (adryianSiarheyeu) [inline src/pages/Pokemons/components/PokemonsLayout/index.js]: this must be separated component (PokemonCard), not bunch of useless markup

**Review by adryianSiarheyeu — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (nbcjoke) [reply src/index.js:15]: Already removed. Tried to fix react dev tools

**Pull request merged**